test/allowlist_for_publicAPI.json

```diff
@@ -369,6 +369,14 @@
     "Any",
     "List"
   ],
+  "torch.ao.sparsity.experimental.data_sparsifier.base_data_sparsifier": [
```

Comment:
No new entry should be added to this file.
In this case, you most likely just want to define `__all__` properly for this submodule.

Reply:
These are generic typing packages [List, Any] that are not actually exposed by the submodule (but used by them). Should this still be inside __all__?

Reply:
I meant that you should have an `__all__` that contain all the public APIs you want to expose and these should not be there.
if you do that, you won't have to add anything to this allowlist.

Reply:
Things like `Any` or `List` are not part of the things your module is exposing publicly (I hope) so these should not be in the `__all__`.

Reply:
> 

I think the default behavior in Python is that any imports within a package would be exposed globally to the user. Please, correct me. Here is a basic example that I have, in case it helps understand what I mean: https://colab.research.google.com/drive/1q-p8zfv_p7Fg1AxR4gWveQszNm-ENCZu?usp=sharing

Reply:
Yes everything is exposed. but there is nothing we can do about that (unless we force everyone to `del Any, List` etc at the end of every single python file). That's why our public API defition is based on `__all__` when it exists.
Note that when you do `from foo import *`, the `__all__` is what will control what gets imported. So in that sense it does control what gets exposed to the user.

Reply:
added __all__ into the .py file